wandb/sdk/internal/system/assets/gpu_apple.py

```diff
@@ -49,35 +73,33 @@ class GPUAppleStats:
 
     def __init__(self) -> None:
         self.samples = deque()
-        self.binary_path = (
-            pathlib.Path(sys.modules["wandb"].__path__[0]) / "bin" / "apple_gpu_stats"
-        ).resolve()
+        self.binary_path = get_apple_gpu_path()
 
     def sample(self) -> None:
         try:
-            command = [str(self.binary_path), "--json"]
-            output = (
-                subprocess.check_output(command, universal_newlines=True)
-                .strip()
-                .split("\n")
-            )[0]
-            raw_stats = json.loads(output)
+            raw_stats = sample_apple_gpu(str(self.binary_path))
+            temp, count = 0, 0
+            for k in ["m1Gpu1", "m1Gpu2", "m1Gpu3", "m1Gpu4", "m2Gpu1", "m2Gpu2"]:
+                if raw_stats.get(k, 0) > 0:
+                    temp += raw_stats[k]
+                    count += 1
```

Comment:
what's `temp`? I can't tell what this is computing

also, it looks like the possibilities here are hardcoded, is there any way to avoid it?

Reply:
we need to fix the swift code... i just copied the implementation from nexus

Reply:
it's temperature